flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -142,6 +142,39 @@ public RocksIncrementalSnapshotStrategy(
         this.localDirectoryName = backendUID.toString().replaceAll("[\\-]", "");
     }
 
+    public RocksIncrementalSnapshotStrategy(
```

Comment:
Can we use telescopic constructor here?
Or even better replace the old constructor with the new one, moving uploader creation to the outside?

Reply:
I also think that the construction of `RocksIncrementalSnapshotStrategy` and `uploader` is a little too tightly coupled, the current implementation is not so flexible. I try it separately.